site-manage/datemanager/impl/src/java/org/sakaiproject/datemanager/impl/DateManagerServiceImpl.java

```diff
@@ -437,11 +454,26 @@ public DateManagerValidation validateAssessments(String siteId, JSONArray assess
 				if (openDate == null) {
 					errored = errors.add(new DateManagerError("open_date", rb.getString("error.open.date.not.found"), "assessments", toolTitle, idx));
 				}
-				if (dueDate == null) {
-					errored = errors.add(new DateManagerError("due_date", rb.getString("error.due.date.not.found"), "assessments", toolTitle, idx));
-				}
-				if (acceptUntil == null && lateHandling) {
-					errored = errors.add(new DateManagerError("accept_until", rb.getString("error.accept.until.not.found"), "assessments", toolTitle, idx));
+				if (acceptUntil != null) {
+					if (dueDate == null) {
+						errors.add(
+								new DateManagerError(
+										"accept_until",
+										"You need to set a valid Due Date if the assessment has a Late Acceptance Date.",
+										"assessments",
+										toolTitle,
+										idx));
+						errored = true;
+					} else if (acceptUntil.isAfter(dueDate)) {
+						errors.add(
+								new DateManagerError(
+										"accept_until",
+										"The Late Submission Date cannot be earlier than the Available Date or the Due Date.",
```

Comment:
Please use rb.getString with a property for i18n

Reply:
Moved the error messages to properties file.